src/addon_transport/http_transport/legacy/mod.rs

```diff
@@ -199,10 +200,27 @@ fn build_legacy_req(transport_url: &Url, path: &ResourcePath) -> Result<Request<
             query.insert("type".into(), serde_json::Value::String(r#type.to_owned()));
             build_jsonrpc("stream.find", json!({ "query": query }))
         }
-        "subtitles" => build_jsonrpc(
-            "subtitles.find",
-            json!({ "query": json!({ "itemHash": id.replace(':', " ") }) }),
-        ),
+        "subtitles" => {
+            let mut query = HashMap::new();
+            query.insert("itemHash", serde_json::Value::String(id.replace(':', " ")));
+            let video_hash = path.get_extra_first_value(VIDEO_HASH_EXTRA_PROP.name.as_str());
+            if let Some(video_hash) = video_hash {
+                query.insert(
+                    VIDEO_HASH_EXTRA_PROP.name.as_str(),
+                    serde_json::Value::String(video_hash.to_owned()),
+                );
+            }
+            let video_size = path.get_extra_first_value(VIDEO_SIZE_EXTRA_PROP.name.as_str());
+            if let Some(video_size) = video_size {
+                query.insert(
+                    VIDEO_SIZE_EXTRA_PROP.name.as_str(),
+                    serde_json::Value::Number(
+                        video_size.parse().expect("Failed to parse videoSize"),
```

Comment:
I'd say that it's better to fail in this case as this is not expected, since the under normal circumstances it can't happened, as the `VideoParams` struct has videoSize as `uint`. And if we just silently swallow and just not send it, we might have a bug that we will ignore and not notice for a long time.

Reply:
And if you have that bug - app crashes and you cannot use it.

Reply:
We should log such errors but we must not crash the app if there's a bug or bad input

Reply:
I'd still think `expect` is better here, because this is a controlled flow only done within the `stremio-core` codebase as the input will always be a number, but fair enough.